testdata/testoutput2

```diff
@@ -19612,6 +19612,693 @@ Failed: error -49 at offset 3 in replacement: unknown substring
 /a(?C)b/never_callout
 Failed: error 203 at offset 3: using callouts is disabled by the application
 
+# --------------
+
+# EXTENDED CHARACTER CLASSES (UTS#18)
+
+/[a[]/
+    [
+ 0: [
+
+/[a[]/alt_extended_class
+Failed: error 106 at offset 4: missing terminating ] for character class
+
+/[a[B]/alt_extended_class
+Failed: error 212 at offset 5: missing terminating ] for character class (note '[' must be escaped under PCRE2_ALT_EXTENDED_CLASS)
+
+/[a[B]]C/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[a]
+          cls:[B]
+          op: ||
+        ]
+        C
+        Ket
+        End
+------------------------------------------------------------------
+    aC
+ 0: aC
+    BC
+ 0: BC
+\= Expect no match
+    [C
+No match
+
+/[[A][B]]/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[A]
+          cls:[B]
+          op: ||
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    A
+ 0: A
+    B
+ 0: B
+\= Expect no match
+    [
+No match
+    ]
+No match
+
+/[[A]||[B]]/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[A]
+          cls:[B]
+          op: ||
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    A
+ 0: A
+    B
+ 0: B
+\= Expect no match
+    C
+No match
+
+/[[^A][B]]/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[\x00-@B-\xff] (neg)
+          cls:[B]
+          op: ||
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    B
+ 0: B
+    C
+ 0: C
+\= Expect no match
+    A
+No match
+
+/[^[A][B]]/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[A]
+          cls:[B]
+          op: ||
+          op: ^
```

Comment:
Even if ^ is a modifier, it is an operator in practice. Btw is `[^[^a-z]]` optimized to `[a-z]`?

Reply:
Ah I see. In the grammar, it's not an operator, because it's attached to the `[^`, so it's parsed as a special form of "open paren". But in the evaluator, it is an operator. The close-paren "]" emits a final OP_ECLASS_NOT if it was opened with a "[^".

> Btw is [^[^a-z]] optimized to [a-z]?

Not yet, no. I _will_ do optimisations (although these are pathological cases you suggest...). If possible I'll do it in a PR next week, although I can include it in the current PR if you require.

Reply:
There are no requirements. Since I prefer to have these optimizations, I should do them.

Reply:
I really will do them, I'd just rather split into a PR next week, as an extra piece of work.

Just give me a few days to get onto it. I'm away this weekend.